tests/util_tests/test_pgvector_migrator.py

```diff
@@ -0,0 +1,806 @@
+                    "embedding": str(sample_embeddings),
+                    "document": f"content_{row_num}",
+                    "cmetadata": json.dumps(
+                        self._create_metadata_for_collection(
+                            collection_name, row_num=row_num, num_cols=num_cols
+                        )
+                    ),
+                },
+            )
+
+    async def _create_pgvector_tables(
+        self,
+        engine: AlloyDBEngine,
+        sample_embeddings: List[float],
+        num_rows: int = 2,
+        num_collections: int = 1,
+        num_cols: int = 3,
+    ) -> None:
+        """Create embeddings as well as collections table."""
+        for collection_num in range(num_collections):
+            collection_name = f"collection_{collection_num}_{COLLECTION_NAME_SUFFIX}"
+            await self._create_collection(
+                engine,
+                collection_name,
+                sample_embeddings,
+                num_rows=num_rows,
+                num_cols=num_cols,
+            )
+
+    async def test_execute(self, engine):
+        await aexecute(engine, query="SELECT 1")
+
+    #### Async tests
+    async def test_aextract_pgvector_collection_exists(self, engine, sample_embeddings):
+        await self._create_pgvector_tables(engine, sample_embeddings)
+        collection_name = f"collection_0_{COLLECTION_NAME_SUFFIX}"
+        results = await aextract_pgvector_collection(engine, collection_name)
+        expected = [
+            {
+                "id": f"uuid_0_{collection_name}",
+                "collection_id": f"collection_id_{collection_name}",
+                "embedding": str(sample_embeddings).replace(" ", ""),
+                "document": "content_0",
+                "cmetadata": self._create_metadata_for_collection(
+                    collection_name, row_num=0, num_cols=3
+                ),
+            },
+            {
+                "id": f"uuid_1_{collection_name}",
+                "collection_id": f"collection_id_{collection_name}",
+                "embedding": str(sample_embeddings).replace(" ", ""),
+                "document": "content_1",
+                "cmetadata": self._create_metadata_for_collection(
+                    collection_name, row_num=1, num_cols=3
+                ),
+            },
+        ]
+        for row in results:
+            assert row in expected
+        assert len(results) == 2
+
+        await aexecute(engine, f"TRUNCATE TABLE {EMBEDDINGS_TABLE}")
+        await aexecute(engine, f"TRUNCATE TABLE {COLLECTIONS_TABLE}")
+
+    async def test_aextract_pgvector_collection_non_existant(self, engine):
+        collection_name = "random_collection"
+        with pytest.raises(ValueError):
+            await aextract_pgvector_collection(engine, collection_name)
+        await aexecute(engine, f"TRUNCATE TABLE {EMBEDDINGS_TABLE}")
+        await aexecute(engine, f"TRUNCATE TABLE {COLLECTIONS_TABLE}")
```

Comment:
nit: Do we need clean up on these tests?

Reply:
Yes. To ensure that tests work independently, we need to move back the tables to their original state.